rootfs/init

```diff
@@ -92,7 +92,8 @@ echo "$line" >&2
 while true; do
     id=$(ls -t /tmp/qmp/req | head -n 1)
     if [ "$id" = "" ]; then
-        sleep 0.1
+        sleep 0.1; if [ -n "$(ls -t /tmp/qmp/req)" ]; then touch /tmp/qmp/req; fi &
+        inotifywait -e attrib,create,delete,modify,move /tmp/qmp/req > /dev/null 2>&1
```

Comment:
This looks like a race condition on every loop iteration. Probably hard to say how long it's going to take `inotifywait` to start and set up the inotify watches, e.g. during high system load?

`inotifywait` doesn't make it easy to _first_ set up the watches and _then_ check if anything interesting happened in the meantime. The best I can think of is something ugly like

```
inotifywait -m -e moved_to --format %f /tmp/qmp/req 2>&1 |
grep --line-buffered . | {
    read _  # "Setting up watches."
    read _  # "Watches established."
    if id=$(ls -t /tmp/qmp/req | grep -m 1 .); then
        # handle "$id"
    fi

    while read -r id; do
        # handle "$id"
    done
}
```

But I haven't really looked at the script, maybe there's a better way. Or maybe the race doesn't even matter in practice.

Reply:
Yes, I thought this might result in a race condition too, that why I originally added `sleep 0.1; if [ -n "$(ls -t /tmp/qmp/req)" ]; then touch /tmp/qmp/req; fi` (which definitely should be in parentheses), but I think your method is a better way of handling it.

Something like this can be slotted straight into the script:

```
inotifywait -e attrib,create,delete,modify,move /tmp/qmp/req 2>&1 |
grep --line-buffered . | {
    while read -r watch_setup; do
        if [ "$watch_setup" = "Watches established." ]; then
           if [ -n "$(ls -t /tmp/qmp/req)" ]; then
               touch /tmp/qmp/req
           fi
        fi
    done
}
```
I'll keep the `create,delete,modify` watches incase the qmp code is extended in the future.